lib/common/results.c

```diff
@@ -903,6 +903,48 @@ pcmk_gaierror2rc(int gai)
     }
 }
 
+/*!
+ * \brief Map a bz2 return code to the most similar Pacemaker return code
+ *
+ * \param[in] bz2  bz2 return code
+ *
+ * \return Most similar Pacemaker return code
+ */
+int
+pcmk_bz2error2rc(int bz2)
+{
+    switch (bz2) {
+        case BZ_OK:
+        case BZ_RUN_OK:
+        case BZ_FLUSH_OK:
+        case BZ_FINISH_OK:
+        case BZ_STREAM_END:
+            return pcmk_rc_ok;
+
+        case BZ_MEM_ERROR:
+            return ENOMEM;
+
+        case BZ_DATA_ERROR:
+        case BZ_DATA_ERROR_MAGIC:
+        case BZ_UNEXPECTED_EOF:
+            return pcmk_rc_bad_input;
+
+        case BZ_IO_ERROR:
+            return EIO;
+
+        case BZ_OUTBUFF_FULL:
+            return EFBIG;
+
+        case BZ_SEQUENCE_ERROR:
+        case BZ_PARAM_ERROR:
+        case BZ_CONFIG_ERROR:
+            return pcmk_rc_bz2_software;
```

Comment:
I'd use this as default and leave off the specific values to minimize portability concerns

Reply:
Changed to return a default of pcmk_rc_compression and got rid of this last group of error codes.